src/Features/Core/Portable/GenerateConstructorFromMembers/GenerateConstructorFromMembersCodeRefactoringProvider.cs

```diff
@@ -42,22 +93,99 @@ internal partial class GenerateConstructorFromMembersCodeRefactoringProvider : A
+
+            result.Add(new FieldDelegatingCodeAction(this, document, state));
             if (state.DelegatedConstructor != null)
             {
-                yield return new ConstructorDelegatingCodeAction(this, document, state);
+                result.Add(new ConstructorDelegatingCodeAction(this, document, state));
+            }
+
+            return result.ToImmutableAndFree();
+        }
+
+        private class GenerateConstructorCodeAction : CodeActionWithOptions
+        {
+            private readonly Document _document;
+            private readonly INamedTypeSymbol _containingType;
+            private readonly GenerateConstructorFromMembersCodeRefactoringProvider _service;
+            private readonly TextSpan _textSpan;
+            private readonly ImmutableArray<ISymbol> _viableMembers;
+
+            public override string Title => FeaturesResources.Generate_constructor;
+
+            public GenerateConstructorCodeAction(
+                GenerateConstructorFromMembersCodeRefactoringProvider service,
+                Document document, TextSpan textSpan,
+                INamedTypeSymbol containingType,
+                ImmutableArray<ISymbol> viableMembers)
+            {
+                _service = service;
+                _document = document;
+                _textSpan = textSpan;
+                _containingType = containingType;
+                _viableMembers = viableMembers;
+            }
+
+            public override object GetOptions(CancellationToken cancellationToken)
+            {
+                var workspace = _document.Project.Solution.Workspace;
+                var service = workspace.Services.GetService<IPickMembersService>();
+                return service.PickMembers(
+                    FeaturesResources.Pick_members_to_be_used_as_constructor_parameters, _viableMembers);
+            }
+
+            protected override async Task<IEnumerable<CodeActionOperation>> ComputeOperationsAsync(
+                object options, CancellationToken cancellationToken)
+            {
+                var result = (PickMembersResult)options;
+                if (result.IsCanceled)
+                {
+                    return ImmutableArray<CodeActionOperation>.Empty;
+                }
+
+                var state = State.Generate(
+                    _service, _document, _textSpan, _containingType, 
+                    result.Members, cancellationToken);
+
+                // There was an existing constructor that matched what the user wants to create.
+                // Generate it if it's the implicit, no-arg, constructor, otherwise just navigate
+                // to the existing constructor
+                if (state.MatchingConstructor != null)
+                {
+                    if (state.MatchingConstructor.IsImplicitlyDeclared)
+                    {
+                        var codeAction = new FieldDelegatingCodeAction(_service, _document, state);
```

Comment:
Can we call this `FieldAssigningCodeAction` or something? Nothing is being delegated through a field or anything like that.